proposals/0014-wave-size-range.md

```diff
@@ -0,0 +1,283 @@
+ possible even if another size is available.
+
+The platform's supported wave size range should first be queried using
+ `WaveLaneCountMin` and `WaveLaneCountMax` from the
+ `D3D12_FEATURE_DATA_D3D12_OPTIONS1` D3D structure.
+This platform range should be used to craft a shader requested range that
+ verifies that the platform accepts and rejects all the shaders it should.
+
+Ensure that the following shader range with platform range combinations are
+ accepted:
+
+* The shader range minimum is the same as the platform range maximum.
+* The shader range maximum is the same as the platform range minimum.
+* The shader range is a superset of the platform range, but only by one power of
+  two value if possible.
+* The shader range is the full 4-128.
+
+Ensure that the following shader range with platform combinations are rejected:
+
+* The shader range minimum is one power of two greater than the platform
+  range maximum.
+* The shader range maximum is one power of two less than the platform range
+  minimum.
+
+For platforms that support more than one wave size, platform treatment of the
+ preferred wave size is needed.
+When available to the platform, the preferred value must be used.
+When not available, but others in the range are, the shader should still be
+ accepted.
+The wave size used can be queried using `WaveGetLaneCount` and fed back to the
+ test to determine that the preferred wave size was used.
+
+For each wave size in the platform range,
+ ensure that the following shader range, preferred value, and platform range
+ combinations are accepted and use the preferred value:
+
+* The the preferred value and shader maximum is equal to the current platform
+  range value and the shader minimum is one power of two less than the platform
+  minimum.
+* The the preferred value and shader minimum is equal to the current platform
+  range value and the shader maximum is one power of two more than the platform
+  maximum.
+* The shader range is the full 4-128 and the preferred value is the current
+  platform range value.
+
+For each wave size in the platform range,
+ ensure that the following shader range, preferred value, and platform range
+ combinations are accepted:
+
+* The the shader maximum is equal to the current platform range value and the
+  preferred value and shader minimum is one power of two less than the platform
+  minimum.
+* The the shader minimum is equal to the current platform range value and the
+  preferred value and shader maximum is one power of two more than the platform
+  maximum.
+* The shader range is the full 4-128 and the preferred value is one power of two
+  less than the current platform range value.
+* The shader range is the full 4-128 and the preferred value is one power of two
+  greater than the current platform range value.
+
+## Alternatives considered
+
+The preferred wave size is seen as useful, but unlike the simple range,
+ it requires adding new runtime data.
```

Comment:
As pointed out by Amar above, I don't think it requires any additional runtime data, just updated metadata in DXIL, but exposing a range already requires this as well.